SRVOCF-303: Add docs on the function annotations feature (#34413)
openshift/openshift-docs: modules/serverless-functions-adding-annotations.adoc

[id="serverless-functions-adding-annotations_{context}"]
= Adding annotations to a function

.Procedure

. Open the `func.yaml` file for your function.

. For every annotation that you want to add, add the following YAML to the `annotations` section:
+
[source,yaml]
----
name: test
namespace: ""
runtime: go
...
annotations:
  <annotation_name>: "<annotation_value>" <1>
----
<1> Substitute `<annotation_name>: "<annotation_value>"` with your annotation.

> 

Reviewer: Hello @rh-max . I think some concrete example would be handy, just as Ashleigh and Lance already suggested. wdyt?

Author: You are right, I've added @lance 's example with Alice.

// Module included in the following assemblies:
//

> // * serverless/functions/serverless-functions-annotations.adoc

Reviewer 2: We don't need to include this anymore as part of OCP doc guidelines - it can be removed. I've actively been removing it from existing modules as well when I come across it 😉

> [id="serverless-functions-adding-annotations_{context}"]

Reviewer 2: files now just start with this heading FYI, see above comment

= Adding annotations to a function

.Procedure

. Open the `func.yaml` file for your function.

> . For every annotation that you want to add, add the following YAML to the `annotations` section:

Reviewer 3: ```suggestion
. You can add Kubernetes annotations to a deployed function, by adding them to the `annotations` section in func.yaml. For example, to indicate that a function was authored by Alice, you might include the following annotation.
```

I suggest a little more contextual information with this introduction.

Author: Added this info, albeit slightly differently than you suggested.

+
[source,yaml]
----
name: test
namespace: ""
runtime: go
...
annotations:

>   color: "red"

Reviewer 3: ```suggestion
  author: "alice@example.com"
```

Author: Went with Ashleigh's suggestion, it the resultant step is more concise.

----
+

> Substitute `color: "red"` with your annotation.

Reviewer 2: This would be much better if it's an example of a real annotation someone might use or if we just use generic variables, e.g.
`<annotation_name>: "<annotation_value>"`

Also, use callouts <1> like this to describe things in the code / tell the user to substitute things.

> Substitute `color: "red"` with your annotation.

Reviewer 3: ```suggestion
```
With the above textual changes, this may be unnecessary.

. Save the configuration.

> The next time you propagate the configuration to the cluster, the corresponding service will have the added annotations.

Reviewer 2: ```suggestion
The next time you propagate the configuration to the cluster, the annotations are added to the corresponding service.
```
suggestion for wording in present tense

> The next time you propagate the configuration to the cluster, the corresponding service will have the added annotations.

Reviewer 3: ```suggestion
The next time you deploy your function to the cluster, the annotations are added to the corresponding service.
```

Author: @lance How about keeping "propagate the configuration", because we might be re-deploying the function?

Reviewer 3: > @lance How about keeping "propagate the configuration", because we might be re-deploying the function?

The sentence as it is actually implies a redeployment with the words "the next time you...".

But my real concern is that we haven't really talked about "propagating the configuration" anywhere in any of these documents. We only talk about deploying. So as a new developer, this might confuse me. It may be a common phrase among seasoned Kubernetes developers. But those developers aren't the primary audience. I would not be opposed to this wording if we state somewhere what "propagate the configuration" means (maybe in the `deploy` docs).

Reviewer 2: I'd go with Lance's wording, deploy IMO is more descriptive / easier to understand than "propagate"

Author: Good points, implemented.
